### PR TITLE
Add rudimentary support for using Solana wallets instead of Keypairs.

### PR DESCRIPTION
**Pull request by dankelleher**
Note 1: No support for decryption until wallets support it
Note 2: The typing could be inproved to simplify the code

### CODE REVIEW

File: src/accounts/solana.ts [hunk 1]
```diff
@@ -5,16 +5,30 @@ import * as solanajs from "@solana/web3.js";
 import nacl from "tweetnacl";
 import base58 from "bs58";
 
+export interface SolanaWallet {
+    signMessage(message: Uint8Array): Promise<Uint8Array>;
+    publicKey: solanajs.PublicKey;
+}
```
Comment (BjrInt) [inline src/accounts/solana.ts:11]: This interface is already defined in the official [Solana WalletAdapter](https://solana-labs.github.io/wallet-adapter/) package.

File: src/accounts/solana.ts [hunk 2]
```diff
@@ -74,41 +92,42 @@ export class SOLAccount extends Account {
...
      * @param message The Aleph message to sign, using some of its fields.
      */
-    override Sign(message: BaseMessage): Promise<string> {
+    override async Sign(message: BaseMessage): Promise<string> {
         const buffer = GetVerificationBuffer(message);
 
-        return new Promise((resolve) => {
-            const bufferSignature = nacl.sign.detached(buffer, this.wallet.secretKey);
+        // TODO let's clean this up with a proper interface later, this is just for prototyping
+        let bufferSignature = !!this.keypair ? nacl.sign.detached(buffer, this.keypair.secretKey) : undefined;
```
Comment (BjrInt) [inline src/accounts/solana.ts:107]: Seems bloated, would rather use regular conditions here.

**Comment by dankelleher** [comment]
Related to https://github.com/aleph-im/aleph-sdk-ts/issues/16

File: tests/messages/store/publish.test.ts [hunk 3]
```diff
@@ -30,5 +30,5 @@ describe("Store message publish", () => {
         const expected = "y";
 
         expect(got).toBe(expected);
-    });
+    }, 20000);
```
Comment (BjrInt) [inline tests/messages/store/publish.test.ts:33]: Should it timeout after 20s ?

**Review by BjrInt — COMMENTED** [review]
Hello @dankelleher ,

Thank you for your submission. Seems functional! :+1: 

It is however worth noting that I've opened a [similar branch a few moments before you did](https://github.com/aleph-im/aleph-sdk-ts/tree/Wallet/Solana), we might want to join forces here.

**Comment by BjrInt** [comment]
duplicate of #31

**Pull request closed without merging**